doc/source/reference/c-api/data_memory.rst

```diff
@@ -159,3 +159,87 @@ A better technique would be to use a ``PyCapsule`` as a base object:
         return NULL;
     }
     ...
+
+Memory Allocation
```

Comment:
Can you find a better title pointing to tracing?  (also I guess we use sentence case not title case now)

The readme can stay with a single link to here, or just remove the entire folder.  Duplicating things doesn't seem useful.

Reply:
@seberg I changed the title of the section to include "Example of memory tracing with ``np.lib.tracemalloc_domain``". I hope it is enough. Thanks.